src/services/signatureHelp.ts

```diff
@@ -574,7 +574,7 @@ namespace ts.SignatureHelp {
                 printer.writeList(ListFormat.TypeParameters, args, sourceFile, writer);
             }
         });
-        const parameters = candidateSignature.parameters.map(p => createSignatureHelpParameterForParameter(p, checker, enclosingDeclaration, sourceFile, printer));
+        const parameters = checker.getExpandedParameters(candidateSignature).map(p => createSignatureHelpParameterForParameter(p, checker, enclosingDeclaration, sourceFile, printer));

```

Comment:
Not sure if this matters or not, but `isVariadic` will be true for functions with expanded parameters even though the expanded signature will not always be.  Not sure if this will be an issue, in my similar [PR](https://github.com/Microsoft/TypeScript/pull/30084/files#diff-77414312b9c8809bd4ba9e31e0c5f940L577), I reevaluated whether the signature is variadic based on the last argument. I found this with the fourslash test I added, don't know if any clients of the API will suffer from  `isVariadic` not being entirely acurate in this case.

Reply:
We don't use the `isVariadic` property ourselves but it is part of the public API so we should make sure it is set correctly.